### PR TITLE
[Merged by Bors] - feat: port Data.Real.ENNReal

### PR DESCRIPTION
**Pull request by urkud**
I did some fixes/refactors while working on this PR. I'm going to move them to new PRs soon.

---

- [x] depends on: #2406
- [x] depends on: #2234 

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#2234
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2234~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#2406~~
* ~~leanprover-community/mathlib4#2234~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Real/ENNReal.lean [hunk 1]
```diff
@@ -167,381 +180,296 @@ theorem ofReal_toReal {a : ℝ≥0∞} (h : a ≠ ∞) : ENNReal.ofReal a.toReal
...
-theorem one_eq_coe : 1 = (↑r : ℝ≥0∞) ↔ 1 = r :=
-  coe_eq_coe
+@[simp, norm_cast] theorem one_eq_coe : 1 = (↑r : ℝ≥0∞) ↔ 1 = r := coe_eq_coe
 #align ennreal.one_eq_coe ENNReal.one_eq_coe
 
-@[simp]
-theorem coe_nonneg : 0 ≤ (↑r : ℝ≥0∞) :=
-  coe_le_coe.2 <| zero_le _
+-- porting note: no longer a `simp`, deprecated, protected
+@[deprecated] protected theorem coe_nonneg : 0 ≤ (r : ℝ≥0∞) := zero_le _
```
Comment (jcommelin) [inline Mathlib/Data/Real/ENNReal.lean:333]: Please give an argument to `deprecated` that informs users what they should use instead.

**Review by jcommelin — COMMENTED** [review]
bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**